jobs/secureproxy/templates/config/nginx.conf.erb

```diff
@@ -116,6 +149,53 @@ http {
       add_header X-XSS-Protection $xss_protection always;
       add_header Content-Type $default_content_type always;
 
+      ##
+      # Implement per-domain IP Whitelist
+      ##
+
+      access_by_lua_block {
+        local headers = ngx.req.get_headers()
+
+        -- bail fast if we don't have a whitelist, or it's not a hostname that needs filtering
+        if ip_whitelist_size == 0 then
+          return
+        end
+
+        -- where is the request coming from
+        local source_ip = ngx.var.remote_addr
+
+        -- if we have an X-Forwarded-For header then use the last one in the list instead of remote_addr
+        local xff = headers["X-Forwarded-For"]
+        if xff ~= nil then
+          local xffl, err = ngx_re.split(xff, ", ")
+          if xffl ~= nil then
+            source_ip = xffl[#xffl]
+          end
+        end
+
+        allow, email = tic.check_ingress({
+          whitelist=whitelist,
+          host_whitelist=host_whitelist,
+          path_whitelist=path_whitelist,
+          request_uri=ngx.var.request_uri,
+          headers=headers,
+          source_ip=source_ip
+        })
+
+        if allow then
+          ngx.log(ngx.NOTICE, "PASS ", source_ip, " is in ", email[2], " whitelist (", email, ")")
+          return
+        else
+          -- fail if the request isn't coming from a whitelisted range
+          ngx.log(ngx.NOTICE, "FAIL ", source_ip, " is not in ", email[2], " whitelist (", email, ")")
+          ngx.status = ngx.HTTP_FORBIDDEN
+          ngx.header.content_type = "application/json; charset=utf-8"
+          -- TODO: Should we be more informative in our error message, should we use a stock cloud controller error code
+          ngx.say(cjson.encode({description="Authorization Error", error_code="SecureProxy-NotAuthorized", code=9162}))
```

Comment:
@mogul @brittag 

Thoughts on the error message displayed when this rule is trigged?